graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -56,13 +58,22 @@ def run_query(
                     message=r"^`id` is deprecated, use `element_id` instead$",
                 )
 
-            try:
-                result = session.run(query, params)
-            except Exception as e:
-                if custom_error:
-                    self.handle_driver_exception(session, e)
-                else:
-                    raise e
+            trys = 1
+            while trys < self._MAX_NUM_RETRYS:
+                try:
+                    result = session.run(query, params)
```

Comment:
Line 85
```python
result.to_df()
```

should most likely also be in the try block. As the result cursor is a stream an error could still be returned later down the road. So you should make sure to fully consume the result before you can assume that the transaction has been committed successfully.

Reply:
Didn't know that :O makes sense though!